java/bulk-import/bulk-import-starter/src/main/java/sleeper/bulkimport/starter/executor/BulkImportArguments.java

```diff
@@ -50,21 +51,31 @@ public List<String> constructArgs(String taskId, String jarLocation) {
         return constructArgs(bulkImportJob, taskId, jarLocation);
     }
 
-    public List<String> constructArgs(BulkImportJob bulkImportJob, String taskId, String jarLocation) {
+    public List<String> constructArgs(BulkImportJob bulkImportJob, String taskId, boolean isServerlessJob) {
         Map<String, String> userConfig = bulkImportJob.getSparkConf();
-        LOGGER.info("Using Spark config {}", userConfig);
 
         String className = bulkImportJob.getClassName() != null ? bulkImportJob.getClassName() : instanceProperties.get(BULK_IMPORT_CLASS_NAME);
 
-        List<String> args = Lists.newArrayList("spark-submit", "--deploy-mode", "cluster", "--class", className);
+        List<String> args = new ArrayList<>();
+
+        if (!isServerlessJob) {
+            args = Lists.newArrayList("spark-submit", "--deploy-mode", "cluster");
+        }
+        args.add("--class");
+        args.add(className);
 
         if (null != userConfig) {
             for (Map.Entry<String, String> configurationItem : userConfig.entrySet()) {
                 args.add("--conf");
                 args.add(configurationItem.getKey() + "=" + configurationItem.getValue());
             }
         }
+        LOGGER.info("Using args {}", args.toString());
```

Comment:
There's no need to call toString here. One reason to pass separate parameters to the logger is so that string processing will only happen if the log level is enabled.

Reply:
This has been removed as it was a testing log